src/contracts/middleware/IndexRegistry.sol

```diff
@@ -235,19 +236,18 @@ contract IndexRegistry is IIndexRegistry {
 
     /// @notice Returns the index of the `operatorId` at the given `blockNumber` for the given `quorumNumber`, or max uint32 if the operator is not active in the quorum
     function _getIndexOfOperatorForQuorumAtBlockNumber(bytes32 operatorId, uint8 quorumNumber, uint32 blockNumber) internal view returns(uint32) {
-        OperatorIndexUpdate memory operatorIndexUpdate;
-        // set to max uint32 value to indicate that the operator is not part of the quorum at all, until this is updated in the loop
-        operatorIndexUpdate.index = type(uint32).max; 
+        uint256 operatorIndexHistoryLength = _operatorIdToIndexHistory[operatorId][quorumNumber].length;
         // loop forward through index history to find the index of the operator at the given block number
         // this is less efficient than looping backwards, but is simpler logic and only called in view functions that aren't mined onchain
         for (uint i = 0; i < _operatorIdToIndexHistory[operatorId][quorumNumber].length; i++) {
-            operatorIndexUpdate = _operatorIdToIndexHistory[operatorId][quorumNumber][i];
+            uint256 listIndex = (operatorIndexHistoryLength - 1) - i;
+            OperatorIndexUpdate memory operatorIndexUpdate = _operatorIdToIndexHistory[operatorId][quorumNumber][listIndex];
             if (operatorIndexUpdate.fromBlockNumber <= blockNumber) {
                 return operatorIndexUpdate.index;
             }
```

Comment:
shouldn't we be checking something about the `toBlockNumber` as well, if we want to have the behavior of returning `max uint32 if the operator is not active in the quorum` ?

Reply:
there is no `toBlockNumber` for the given struct. when the operator deregisters, we push an update with `type(uint32).max` as the index

Reply:
I would comment on the struct itself, where it's written, and especially anywhere we use the special value.
Probably also name a constant something like `OPERATOR_DEREGISTERED_INDICATOR` and use that instead of the "magic value" of max(uint32)